lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -809,10 +809,14 @@ def add_stored_credential_using_gateway_specific_fields(xml, options)
         stored_credential_params = generate_stored_credential_params(is_initial_transaction, options[:stored_credential_initiated_reason])
 
         xml.storedCredentials stored_credential_params do
-          xml.schemeTransactionIdentifier options[:stored_credential_transaction_id] if options[:stored_credential_transaction_id] && !is_initial_transaction
+          xml.schemeTransactionIdentifier options[:stored_credential_transaction_id] if options[:stored_credential_transaction_id] && customer_initiated_transaction?(options, is_initial_transaction)
         end
       end
 
+      def customer_initiated_transaction?(options, is_initial_transaction)
+        !is_initial_transaction && options[:stored_credential]&.dig(:initiator) != 'cardholder'
+      end
```

Comment:
I think we can name this method as `subsequent_non_cardholder_transaction?` we could also put `dig` in the `stored_credential` option

```suggestion
      def subsequent_non_cardholder_transaction?(options, is_initial_transaction)
        !is_initial_transaction && options.dig(:stored_credential, :initiator) != 'cardholder'
      end
```

Reply:
Done!